docs/samples/Microsoft.ML.Samples/Dynamic/Trainers/BinaryClassification/FieldAwareFactorizationMachineWithOptions.cs

```diff
@@ -0,0 +1,167 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.ML;
+using Microsoft.ML.Data;
+using Microsoft.ML.Trainers;
+
+namespace Samples.Dynamic.Trainers.BinaryClassification
+{
+    public static class FieldAwareFactorizationMachineWithOptions
+    {// This example first train a field-aware factorization to binary classification, measure the trained model's quality, and finally
+        // use the trained model to make prediction.
+        public static void Example()
+        {
+            // Create a new context for ML.NET operations. It can be used for exception tracking and logging, 
+            // as a catalog of available operations and as the source of randomness.
+            // Setting the seed to a fixed number in this example to make outputs deterministic.
+            var mlContext = new MLContext(seed: 0);
+
+            // Create a list of training data points.
+            IEnumerable<DataPoint> data = GenerateRandomDataPoints(500);
+
+            // Convert the list of data points to an IDataView object, which is consumable by ML.NET API.
+            var trainingData = mlContext.Data.LoadFromEnumerable(data);
+
+            // Define trainer options.
+            var options = new FieldAwareFactorizationMachineTrainer.Options
+            {
+                FeatureColumnName = nameof(DataPoint.Field0),
+                ExtraFeatureColumns = new[] { nameof(DataPoint.Field1), nameof(DataPoint.Field2) },
+                LabelColumnName = nameof(DataPoint.Label),
+                LambdaLatent = 0.01f,
+                LambdaLinear = 0.001f,
+                LatentDimension = 16,
+                NumberOfIterations = 50,
+                LearningRate = 0.5f
+            };
+
+            // Define the trainer.
+			// This trainer trains field-aware factorization (FFM) for binary classification. See https://www.csie.ntu.edu.tw/~cjlin/papers/ffm.pdf
+            // for the theory behind and https://github.com/wschin/fast-ffm/blob/master/fast-ffm.pdf for the training
+            // algorithm implemented in ML.NET.
+            var pipeline = mlContext.BinaryClassification.Trainers.FieldAwareFactorizationMachine(options);
+
+            // Train the model.
+            var model = pipeline.Fit(trainingData);
+
+            // Run the model on training data set.
+            var transformedTrainingData = model.Transform(trainingData);
+
+			// Measure the quality of the trained model.
+            var metrics = mlContext.BinaryClassification.Evaluate(transformedTrainingData);
+
+			// Show the quality metrics.
+            Microsoft.ML.SamplesUtils.ConsoleUtils.PrintMetrics(metrics);
+
+            // Expected output:
+            //   Accuracy: 0.99
+            //   AUC: 1.00
```

Comment:
>1.00 [](start = 22, length = 4)

[optional] can we make it a little harder for the classifier? maybe reduce the label delta from 0.2f to 0.05f or some lower number?

Reply:
Maybe no? It's super important to show a case where a model works well. This is a motive way to push user to think about feature engineering if my model doesn't work on their case.